engine/src/flutter/display_list/utils/dl_matrix_clip_tracker.h

```diff
@@ -49,18 +49,25 @@ class DisplayListMatrixClipState {
   bool is_cull_rect_empty() const { return cull_rect_.IsEmpty(); }
 
   void translate(DlScalar tx, DlScalar ty) {
+    // No impact on translate scale.
     matrix_ = matrix_.Translate({tx, ty});
   }
   void scale(DlScalar sx, DlScalar sy) {
+    // No impact on translate scale.
     matrix_ = matrix_.Scale({sx, sy, 1.0f});
   }
   void skew(DlScalar skx, DlScalar sky) {
+    is_translate_scale_ = false;
     matrix_ = matrix_ * DlMatrix::MakeSkew(skx, sky);
   }
   void rotate(DlRadians angle) {
+    is_translate_scale_ = false;
```

Comment:
90 degree rotations might be common and they can be optimized as well, but it isn't as simple as the new method. Also, unfortunately, rotate(radians) is hard to detect a quadrant rotation because the quadrant angles are all transcendental by being related to pi.